homie-influx/src/main.rs

```diff
@@ -0,0 +1,388 @@
+            name: None,
+            datatype: Some(Datatype::Integer),
+            settable: false,
+            retained: true,
+            unit: None,
+            format: None,
+            value: Some("42".to_owned()),
+        };
+        assert_debug_eq(
+            influx_value_for_homie_property(&property).unwrap(),
+            Value::Integer(42),
+        );
+    }
+
+    #[test]
+    fn influx_value_for_float() {
+        let property = Property {
+            id: "property_id".to_owned(),
+            name: None,
+            datatype: Some(Datatype::Float),
+            settable: false,
+            retained: true,
+            unit: None,
+            format: None,
+            value: Some("42.3".to_owned()),
+        };
+        assert_debug_eq(
+            influx_value_for_homie_property(&property).unwrap(),
+            Value::Float(42.3),
+        );
+    }
+
+    #[test]
+    fn influx_value_for_boolean() {
+        let property = Property {
+            id: "property_id".to_owned(),
+            name: None,
+            datatype: Some(Datatype::Boolean),
+            settable: false,
+            retained: true,
+            unit: None,
+            format: None,
+            value: Some("true".to_owned()),
+        };
+        assert_debug_eq(
+            influx_value_for_homie_property(&property).unwrap(),
+            Value::Boolean(true),
+        );
+    }
+
+    #[test]
+    fn influx_value_for_string() {
+        let property = Property {
+            id: "property_id".to_owned(),
+            name: None,
+            datatype: Some(Datatype::String),
+            settable: false,
+            retained: true,
+            unit: None,
+            format: None,
+            value: Some("abc".to_owned()),
+        };
+        assert_debug_eq(
+            influx_value_for_homie_property(&property).unwrap(),
+            Value::String("abc".to_owned()),
+        );
+    }
+
+    #[test]
+    fn influx_value_for_enum() {
+        let property = Property {
+            id: "property_id".to_owned(),
+            name: None,
+            datatype: Some(Datatype::Enum),
+            settable: false,
+            retained: true,
+            unit: None,
+            format: None,
+            value: Some("abc".to_owned()),
+        };
+        assert_debug_eq(
+            influx_value_for_homie_property(&property).unwrap(),
+            Value::String("abc".to_owned()),
+        );
+    }
+
+    #[test]
+    fn influx_value_for_color() {
+        let property = Property {
+            id: "property_id".to_owned(),
+            name: None,
+            datatype: Some(Datatype::Color),
+            settable: false,
+            retained: true,
+            unit: None,
+            format: None,
+            value: Some("12,34,56".to_owned()),
+        };
+        assert_debug_eq(
+            influx_value_for_homie_property(&property).unwrap(),
+            Value::String("12,34,56".to_owned()),
```

Comment:
I was about to suggest splitting value into red green and blue components, but maybe a string is what we really mean. In practice you don't use all 256^3 colours for LED light: you just use one of about 10 specific value combinations from a colour palette.